module/profiler/uploader.go

```diff
@@ -0,0 +1,121 @@
+package profiler
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/rs/zerolog"
+	"google.golang.org/api/option"
+	gtransport "google.golang.org/api/transport/grpc"
+	pb "google.golang.org/genproto/googleapis/devtools/cloudprofiler/v2"
+	"google.golang.org/grpc"
+)
+
+const (
+	apiAddress = "cloudprofiler.googleapis.com:443"
+	scope      = "https://www.googleapis.com/auth/monitoring.write"
+
+	maxMsgSize = 1 << 30 // 1GB
+)
+
+type NoopUploader struct{}
+
+func (u *NoopUploader) Upload(ctx context.Context, filename string, pt pb.ProfileType) error {
+	return nil
+}
+
+type Uploader interface {
+	Upload(ctx context.Context, filename string, pt pb.ProfileType) error
+}
+
+type Params struct {
+	ProjectID string
+	ChainID   string
+	Role      string
+	Version   string
+	Commit    string
+	Instance  string
+}
+
+type UploaderImpl struct {
+	log    zerolog.Logger
+	client pb.ProfilerServiceClient
+
+	ProjectId  string
+	Deployment *pb.Deployment
+}
+
+func NewUploader(log zerolog.Logger, params Params, opts ...option.ClientOption) (Uploader, error) {
+	log = log.With().Str("component", "profile_uploader").Logger()
+
+	defaultOpts := []option.ClientOption{
+		option.WithEndpoint(apiAddress),
+		option.WithScopes(scope),
+		option.WithUserAgent("OfflinePprofUploader"),
+		option.WithTelemetryDisabled(),
+		option.WithGRPCDialOption(
+			grpc.WithDefaultCallOptions(
+				grpc.MaxCallSendMsgSize(maxMsgSize),
```

Comment:
you realize this potentially means a 1GB protobuf message, right?

Reply:
yep. Our traces right now are about ~60MiB, so i've given it ~ an order of magnitude of potential growth.  That said, it is still an improvement to the default 2GiB: https://github.com/grpc/grpc-go/blob/13b378bc458535e6da1693fc67e09a4951cd3259/clientconn.go#L97

Reply:
o.  interesting.  this default used to be 4MB a few years back (I had to bump it up to 6mb by default at dbx cuz of block size).  i guess everyone got tired of manually increasing it